### PR TITLE
Send change notifications in grey_replaceText

### PR DESCRIPTION
**Pull request by Arcank**
When setting the text property directly, neither UITextFieldTextDidChangeNotification nor UIControlEventEditingChanged are sent.
Since the goal of grey_replaceText is to simulate a user entry, the notifications should be sent manually.

Closes https://github.com/google/EarlGrey/issues/253.

### CODE REVIEW

File: EarlGrey/Action/GREYActions.m [hunk 1]
```diff
@@ -330,6 +330,16 @@ + (void)grey_webSetText:(id)element text:(NSString *)text {
       [GREYActions grey_webSetText:element text:text];
     } else {
       [element setText:text];
+            // For UIControl-s, send actions for UIControlEventEditingChanged, as it is not sent when directly
+      // setting the text.
+      [element sendActionsForControlEvents:UIControlEventEditingChanged];
+      // For UITextField-s, send the UITextFieldTextDidChangeNotification notification, as it is not sent
```
Comment (bootstraponline) [inline EarlGrey/Action/GREYActions.m]: I think the code is cleaner without these comments since they just describe what the code is doing.

Comment (Arcank) [inline EarlGrey/Action/GREYActions.m]: Ok. The PR comment will suffice to describe the "why".

File: EarlGrey/Action/GREYActions.m [hunk 2]
```diff
@@ -330,6 +330,12 @@ + (void)grey_webSetText:(id)element text:(NSString *)text {
       [GREYActions grey_webSetText:element text:text];
     } else {
       [element setText:text];
+      [element sendActionsForControlEvents:UIControlEventEditingChanged];
```
Comment (khandpur) [inline EarlGrey/Action/GREYActions.m]: @Arcank do we need to also do this for setting the text directly on web elements?

**Comment by tirodkar** [comment]
@Arcank, I just noticed that there's no tests for the notification here. Can you please add a test here - https://github.com/google/EarlGrey/blob/c1b55e6da04816c3739aef5658277495159b8ba7/Tests/FunctionalTests/Sources/FTRSwiftTests.swift#L47 ??

File: EarlGrey/Action/GREYActions.m [hunk 3]
```diff
@@ -330,6 +330,12 @@ + (void)grey_webSetText:(id)element text:(NSString *)text {
       [GREYActions grey_webSetText:element text:text];
     } else {
       [element setText:text];
```
Comment (tirodkar) [inline EarlGrey/Action/GREYActions.m]: Add a check here for when the text already matches the text present.

Comment (bootstraponline) [inline EarlGrey/Action/GREYActions.m]: > Add a check here for when the text already matches the text present.

Why do we care? setText is overriding, not appending.

File: EarlGrey/Action/GREYActions.m (hunk 2 above)
Comment (Arcank) [inline EarlGrey/Action/GREYActions.m]: Web elements are not UIControl-s.

**Comment by Arcank** [comment]
Please take another look. Thank you!

File: Tests/FunctionalTests/Sources/FTRSwiftTests.swift [hunk 4]
```diff
@@ -153,4 +159,24 @@ class FunctionalTestRigSwiftTests: XCTestCase {
...
   }
+
+  func registerForChanges() -> GREYActionBlock {
+    NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(textFieldChanged), name: UITextFieldTextDidChangeNotification, object: nil)
+    return GREYActionBlock.actionWithName("Register to editing events") {
+      (element: AnyObject!, _: UnsafeMutablePointer<NSError?>) -> Bool in
+      element.addTarget(self, action: #selector(self.editingChanged), forControlEvents: .EditingChanged)
+      return true
+    }
+
```
Comment (sid-github) [inline Tests/FunctionalTests/Sources/FTRSwiftTests.swift]: Remove blank line

File: EarlGrey/Action/GREYActions.m (hunk 2 above)
Comment (sid-github) [inline EarlGrey/Action/GREYActions.m]: Is this to send the "UITextFieldTextDidBeginEditingNotification"?

File: Tests/FunctionalTests/Sources/FTRSwiftTests.swift (hunk 4 above)
Comment (Arcank) [inline Tests/FunctionalTests/Sources/FTRSwiftTests.swift]: Done. I think I opened the PR to changes by contributors. That should cut some back and forth, especially as I am based in France :)

File: EarlGrey/Action/GREYActions.m (hunk 2 above)
Comment (Arcank) [inline EarlGrey/Action/GREYActions.m]: No, it's a different notification.

Adding DidBegin, DidEndOnExit and DidEnd.

**Comment by sid-github** [comment]
LGTM

File: EarlGrey/Action/GREYActions.m (hunk 2 above)
Comment (tirodkar) [inline EarlGrey/Action/GREYActions.m]: We'll get this in the next release since you're adding it. Thanks.

**Comment by Arcank** [comment]
Added more notifications. Please advise if still LGTY. Thank you!

**Comment by tirodkar** [comment]
@Arcank, I've updated the code with another check and some formatting.

**Comment by khandpur** [comment]
Trying to resolve this failure:
https://travis-ci.org/google/EarlGrey/jobs/159714700

before merging this in.

**Comment by khandpur** [comment]
LGTM

**Pull request merged**